pkg/storage/bulk/sst_batcher.go

```diff
@@ -228,22 +264,47 @@ type sstSpan struct {
 // SST spans a split, in which case it is iterated and split into two SSTs, one
 // for each side of the split in the error, and each are retried.
 func AddSSTable(
-	ctx context.Context, db sender, start, end roachpb.Key, sstBytes []byte, disallowShadowing bool,
+	ctx context.Context,
+	db sender,
+	start, end roachpb.Key,
+	sstBytes []byte,
+	disallowShadowing bool,
+	ms *enginepb.MVCCStats,
 ) error {
-	// Create an iterator that iterates over the top level SST for use in stats
-	// computation and to produce split ssts if needed for retries.
+	now := timeutil.Now().UnixNano()
 	iter, err := engine.NewMemSSTIterator(sstBytes, true)
 	if err != nil {
 		return err
 	}
 	defer iter.Close()
 
-	now := timeutil.Now().UnixNano()
-	stats, err := engine.ComputeStatsGo(
-		iter, engine.MVCCKey{Key: start}, engine.MVCCKey{Key: end}, now,
-	)
-	if err != nil {
-		return errors.Wrapf(err, "computing stats for SST [%s, %s)", start, end)
+	var stats enginepb.MVCCStats
+	fast := ms != nil
+	if fast {
+		stats = *ms
+		stats.LastUpdateNanos = now
+	}
+
+	// If we can't use the fast stats path, or race test is enabled,
+	// compute stats over the SST being ingested.
+	if !fast || util.RaceEnabled {
```

Comment:
I was imagining we'd do this in cmd_add_sstable where we choose to use the client-provided stats vs out own instead of here so it'd run no matter who/how we sent an SST, but either way works.